dipy/tracking/utils.py

```diff
@@ -503,6 +515,11 @@ def random_seeds_from_mask(mask, seeds_per_voxel=1, affine=None):
     seeds = where + grid - .5
     seeds = asarray(seeds)
 
+    if not seed_count_per_voxel:
+        # Randomize the seeds and select the requested amount
+        np.random.shuffle(seeds)
```

Comment:
Wouldn't be better to create its own random generator using a seed (ideally provided as an argument to this function) instead of relying on the user having to call `np.random.seed(seed)`. I think changing the seed of the global random generator (i.e. the one in `np.random`) is prone unreproducible results. (e.g., for some reason you decided to call `np.random.rand()` in another part of your code before calling this function).